src/module_wrap.cc

```diff
@@ -584,7 +594,15 @@ void ModuleWrap::Link(const FunctionCallbackInfo<Value>& args) {
 
     ModuleCacheKey module_cache_key = ModuleCacheKey::From(
         context, requests->Get(context, i).As<ModuleRequest>());
-    dependent->resolve_cache_[module_cache_key].Reset(isolate, module_object);
+
+    if (resolve_cache.find(module_cache_key) != resolve_cache.end() &&
```

Comment:
If we dedupe requests, am I right in thinking this is no longer possible?

Reply:
If the module requests are dedup-ed ignoring `phase`, yes, this is not possible.